rust/objectstore_ext/src/s3.rs

```diff
@@ -214,6 +215,31 @@ impl ObjectStoreFactory {
             )),
         }
     }
+
+    fn wrap_s3_store(&self, store: AmazonS3, bucket: &str) -> Arc<dyn ObjectStore> {
+        if self.use_readahead {
+            Arc::new(apply_s3_logging_store(
+                apply_readahead_store(store, bucket),
+                bucket,
+            ))
+        } else {
+            Arc::new(apply_s3_logging_store(store, bucket))
+        }
+    }
+}
+
+fn apply_readahead_store(store: AmazonS3, bucket: &str) -> impl ObjectStore {
```

Comment:
This function shouldn't be specific to AmazonS3. It should take an `impl ObjectStore`

Reply:
More idiomatically, define as:
```
fn apply_readahead_store<T: ObjectStore>(store: T, bucket: &str) -> ReadaheadStore<T> {
```